### PR TITLE
Profiling additions

### PR DESCRIPTION
**Pull request by steven-murray**
Adds profiling option to the `hera-sim-vis.py` script, which makes it possible to do line-profiling as part of the run.

### CODE REVIEW

File: scripts/hera-sim-vis.py [hunk 1]
```diff
@@ -132,6 +136,16 @@ def cprint(*args, **kwargs):
         for fnc in simulator._functions_to_profile:
             profiler.add_function(fnc)
 
+        # Now add any user-defined functions that they want to be profiled.
+        # Functions must be sent in as "path.to.module:function_name" or
+        # "path.to.module:Class.method".
```
Comment (r-pascua) [inline scripts/hera-sim-vis.py:141]: Can you add the part about the required format to the `help` info for the `profile_funcs` argument?

**Review by r-pascua — CHANGES_REQUESTED** [review]
Looks good to me! Just expose the required format for the `profile_funcs` to the user and I'll approve. (We'll also want to make a note about this in any tutorial that touches on the profiling capabilities.)

File: scripts/hera-sim-vis.py (hunk 1 above)
Comment (steven-murray) [reply scripts/hera-sim-vis.py:141]: Ah yeah, good idea.

**Review by r-pascua — APPROVED** [review]
Thanks for making the change! It all looks good to me!

**Pull request merged**